module/solutions/helper.js

```diff
@@ -1,34 +1,195 @@
 /**
  * name : solutions/helper.js
- * author : Akash
- * created-date : 22-feb-2019
+ * author : Praveen
+ * created-date : 13-jul-2024
  * Description : Solution related helper functionality.
  */
 
 //Dependencies
 const entitiesHelper = require(MODULES_BASE_PATH + '/entities/helper');
 const userExtensionHelper = require(MODULES_BASE_PATH + '/userExtension/helper');
 const criteriaHelper = require(MODULES_BASE_PATH + '/criteria/helper');
-const entityTypesHelper = require(MODULES_BASE_PATH + '/entityTypes/helper');
 const userRolesHelper = require(MODULES_BASE_PATH + '/userRoles/helper');
 const appsPortalBaseUrl = process.env.APP_PORTAL_BASE_URL + '/';
+const programsHelper = require(MODULES_BASE_PATH + '/programs/helper');
+const timeZoneDifference = process.env.TIMEZONE_DIFFRENECE_BETWEEN_LOCAL_TIME_AND_UTC;
+const surveyHelper = require(MODULES_BASE_PATH + '/surveys/helper');
+const observationHelper = require(MODULES_BASE_PATH + '/observations/helper');
+const userHelper = require(MODULES_BASE_PATH + '/users/helper');
+const improvementProjectService = require(ROOT_PATH + '/generics/services/improvement-project');
+const validateEntity = process.env.VALIDATE_ENTITIES;
+const programUsersHelper = require(MODULES_BASE_PATH + '/programUsers/helper');
+const programsQueries = require(DB_QUERY_BASE_PATH + '/programs');
+const solutionsQueries = require(DB_QUERY_BASE_PATH + '/solutions');
+const entityManagementService = require(ROOT_PATH + '/generics/services/entity-management');
+
 /**
  * SolutionsHelper
  * @class
  */
 module.exports = class SolutionsHelper {
+  /**
+   * Create solution.
+   * @method
+   * @name createSolution
+   * @param {Object} solutionData - solution creation data.
+   * @param {Boolean} checkDate this is true for when its called via API calls
+   * @returns {JSON} solution creation data.
+   */
+
+  static createSolution(solutionData, checkDate = false) {
```

Comment:
@praveenKDass make sure all newly added function have validator. Check for all module

Reply:
Yes added the validation @VISHNUDAS-tunerlabs